src/main/java/decodes/sql/PlatformListIO.java

```diff
@@ -100,1194 +38,1195 @@
-				}
-				return networkListArray;
-			}
-		}
-		catch(SQLException ex)
-		{
-			warning("SQL Error reading networkList: " + ex);
-			return null;
-		}
-	}
+            // Use transport media to see if this platform already exists in
+            // the database.
+//            int id = getIdByTransportMedia(p);
+//            if (id != Constants.undefinedId)
+//            {
+//                p.setId(id);
+//                update(p);
+//            }
+//            else
+                insert(p);
+        }
+
+        //System.out.println("    num PlatformSensors is " +
+        //    p.platformSensors.size());
+    }
+
+
+    /**
+    * Writes the changed  data for a pre-existing Platform back out to the
+    * database, using an UPDATE query.
+    * @param p the Platform
+    */
+    private void update(Platform p)
+        throws DatabaseException, SQLException
+    {
+        String q =
+            "UPDATE Platform SET " +
+              "Agency = " + sqlString(p.agency) + ", " +
+              "IsProduction = " + sqlString(p.isProduction) + ", " +
+              "SiteId = " + sqlOptHasId(p.getSite()) + ", " +
+              "ConfigId = " + sqlOptHasId(p.getConfig()) + ", " +
+              "Description = " + sqlOptString(p.description) + ", " +
+              "LastModifyTime = " + sqlOptDate(p.lastModifyTime) + ", " +
+              "Expiration = " + sqlOptDate(p.expiration);
+        if (getDatabaseVersion() >= DecodesDatabaseVersion.DECODES_DB_7)
+            q = q + ", platformDesignator = " +
+                sqlOptString(p.getPlatformDesignator());
+
+        q = q + " WHERE ID = " + p.getId();
+
+        try { executeUpdate(q); }
+        catch(Exception ex)
+        {
+            if (ex.toString().toLowerCase().contains("unique"))
+            {
+                // This means a different platform exists with the (siteid,designator)
+                // in the above update.
+                if (isDbImport && DecodesSettings.instance().autoDeleteOnImport)
+                {
+                    q = "select id from platform where siteid = " +
+                        sqlOptHasId(p.getSite()) + " and platformdesignator "
+                        + (p.getPlatformDesignator() == null || p.getPlatformDesignator().length()==0
+                          ? "is null" : ("= " + sqlString(p.getPlatformDesignator())));
+                    try (Statement stmt = createStatement();)
+                    {
+                        try (ResultSet rs = stmt.executeQuery(q);)
+                        {
+                            DbKey pid = Constants.undefinedId;
+                            if (rs != null && rs.next())
+                            {
+                                log.info("Imported platform clashes with old platform ID={}"
+                                     +" -- deleting old platform.",pid);
+                                pid = DbKey.createDbKey(rs, 1);
+                                rs.close();
```

Comment:
these aren't actually necessary. both the statement and result set are already within a try-with.